### PR TITLE
s2i datapower

### PR DESCRIPTION
**Pull request by raffaelespazzoli**
this is an s2i image for the ibm datapower appliance.
It contains the instructions for creating the s2i image, explanation on how to work in the two modes: experimentation and immutable infrastructure and an example

### CODE REVIEW

**Comment by etsauer** [comment]
[DataPower (ESB, API Gateway)](https://trello.com/c/XtIZ3YQW/14-datapower-esb-api-gateway)

File: s2i-datapower/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+This is the project win which we will run the below commands, you can use a different project.
+
+```
+oc new-project datapower
+```
+
+## Create the s2i-datapower Image
+
+This commands will create a build config whose outout will be the s2i-datapower.
```
Comment (dstockdreher) [inline s2i-datapower/README.md]: output?

File: s2i-datapower/README.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
+# Datapower s2i
+
+This project demonstrates how to create an s2i Datapower image to facilitate Datapower deployments in Openshift.
+
+![IBM Datapower](https://avatars3.githubusercontent.com/u/8836442?v=3&s=200 "IBM Datapower")
+
+## Table of content
+
```
Comment (dstockdreher) [inline s2i-datapower/README.md]: Would a prerequisites section be prudent?

File: s2i-datapower/README.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+oc patch dc/datapower --patch '{"spec":{"template":{"spec":{"serviceAccountName": "datapower"}}}}'
+oc set volume dc/datapower --add -m /drouter/config --name datapower-config -t pvc --claim-name=datapower-config --claim-size=1G
+oc set volume dc/datapower --add -m /drouter/local --name datapower-local -t pvc --claim-name=datapower-local --claim-size=1G
+oc create route passthrough datapower-svc --service datapower --port=8080
+oc create route passthrough datapower-console --service datapower --port=9090
+```
+enabling openshift sso for an application
+
+```
+oc create -f ???
```
Comment (dstockdreher) [inline s2i-datapower/README.md]: Not sure what is supposed to be here?  Is there an SSO section for Datapower?  is this just WIP?

File: s2i-datapower/README.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+* [Create the s2i-datapower Image](#create-the-s2i-datapower-image)
+* [The Experimentation Use Case](#the-experimentation-use-case)
+* [The Immutable Use Case](#the-immutable-use-case)
+* [Examples](#examples)
+
+
+
+## Create a Project 
+
+This is the project win which we will run the below commands, you can use a different project.
```
Comment (dstockdreher) [inline s2i-datapower/README.md]: within?

File: s2i-datapower/README.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+```
+oc new-project datapower
+```
+
+## Create the s2i-datapower Image
+
+This commands will create a build config whose outout will be the s2i-datapower.
+
+```
+oc new-build https://github.com/raffaelespazzoli/containers-quickstarts#datapower --context-dir=s2i-datapower --name=s2i-datapower --strategy=docker
```
Comment (sabre1041) [inline s2i-datapower/README.md]: This should reference the [containers-quickstarts](https://github.com/redhat-cop/containers-quickstarts) repository

File: s2i-datapower/README.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+The configurations are save in two persistent storage volumes (/drouter/config and /drouter/local)
+
+The expectation is that the configurations can be retrieved from the persistent volumes and use as an input for the s2i-datapower image in the immutable use case.
+
+This is just one possbile approach to creating configuration. Any approach is fine.
+
+to use the image in this mode issue the following commands
+
+```
+oc new-app --docker-image=datapower/s2i-datapower -e DATAPOWER_ACCEPT_LICENSE=true -e DATAPOWER_WORKER_THREADS=4 --name=datapower
```
Comment (sabre1041) [inline s2i-datapower/README.md]: `new-app` should use `--image-stream=s2i-datapower`

File: s2i-datapower/README.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+oc set volume dc/datapower --add -m /drouter/config --name datapower-config -t pvc --claim-name=datapower-config --claim-size=1G
```
Comment (sabre1041) [inline s2i-datapower/README.md]: Receiving the following error

```
persistentvolumeclaims/datapower-config
error: deploymentconfigs/datapower volume mount '/drouter/config' already exists for container 'datapower'
```

File: s2i-datapower/README.md (hunk 1 above)
Comment (raffaelespazzoli) [inline s2i-datapower/README.md]: fixed

File: s2i-datapower/README.md (hunk 2 above)
Comment (raffaelespazzoli) [inline s2i-datapower/README.md]: what would you want to see in the prerequisite section?

File: s2i-datapower/README.md (hunk 3 above)
Comment (raffaelespazzoli) [inline s2i-datapower/README.md]: yes that was just an abandoned experiment. deleted

File: s2i-datapower/README.md (hunk 4 above)
Comment (raffaelespazzoli) [inline s2i-datapower/README.md]: fixed

File: s2i-datapower/README.md (hunk 5 above)
Comment (raffaelespazzoli) [inline s2i-datapower/README.md]: fixed

File: s2i-datapower/README.md (hunk 6 above)
Comment (raffaelespazzoli) [inline s2i-datapower/README.md]: fixed

File: s2i-datapower/README.md (hunk 7 above)
Comment (raffaelespazzoli) [inline s2i-datapower/README.md]: fixed, I think is has to do with the fact that I did this project a while ago months ago.

**Comment by sabre1041** [comment]
@raffaelespazzoli circling back to this old PR. 

A few items.

1. Received a few warnings when building the datapower image

```
chown: /drouter/image: No such file or directory
chown: /drouter/webgui/firmware.xml: No such file or directory
chgrp: /drouter/image: No such file or directory
chgrp: /drouter/webgui/firmware.xml: No such file or directory
```

2. "mediating cake-php with datapower" example does not appear to exist

How do you want to approach this PR?

**Pull request closed without merging**